queue/abstract.lua

```diff
@@ -152,16 +159,14 @@ function tube.touch(self, id, delta)
 end
 
 function tube.ack(self, id)
-    local _taken = box.space._queue_taken:get{connection.id(), self.tube_id, id}
-    if _taken == nil then
-        error("Task was not taken in the session")
-    end
+    local conn_id = connection.id()
+    check_task_on_taken(self.tube_id, id, conn_id)
     local tube = box.space._queue:get{self.name}
     local space_name = tube[3]
 
     self:peek(id)
     -- delete task
-    box.space._queue_taken:delete{connection.id(), self.tube_id, id}
+    box.space._queue_taken:delete{conn_id, self.tube_id, id}
```

Comment:
Nit: Does not look relevant to this commit.

Reply:
I disagree.Before `connection.id()` used in one place in a function. Now `conn_id` is used in two places and using local variable let to avoid unnecessary call `connection.id()`.